mm2src/mm2_main/tests/mm2_tests/mm2_tests_inner.rs

```diff
@@ -1564,6 +1564,9 @@ fn test_cancel_order() {
     assert_eq!(alice_orderbook.asks.len(), 0, "Alice RICK/MORTY asks are not empty");
 }
 
+/// This also covers recently canceled orders implementation.
+/// The cancellation message sometimes arrives to Alice before the order creation message,
+/// in this case Alice gets the order creation message from p2p cache using IHAVE / IWANT control messages.
 #[test]
 #[cfg(not(target_arch = "wasm32"))]
 fn test_cancel_all_orders() {
```

Comment:
> I think we should **intentionally** send the order cancellation message before creating it.

There is no way to do that intentionally since to cancel an order from the maker node it has to have the order created first, in the current test order cancellation almost always arrive first, I never found a case where it didn't but changed the check for log that the creation message was ignored in case it doesn't happen. But I will try to send the order cancellation message first intentionally in a new test using `maker_order_cancelled_p2p_notify` straight away from ctx, if I succeed will revert the changes to the current test.

Reply:
> But I will try to send the order cancellation message first intentionally in a new test using maker_order_cancelled_p2p_notify straight away from ctx, if I succeed, I will revert the changes to the current test.

Couldn't do it through ctx, but added a test here https://github.com/KomodoPlatform/komodo-defi-framework/pull/2232/commits/598045dbb779a8543294b7183db73ab316e345a7 that is deterministic and cancel message is always received before creation.